pkg/worker/workerhandler/claimresolvehandler/custom_ensure.go

```diff
@@ -17,38 +19,83 @@ import (
 func (h *SystemHandler) Ensure(tas *task.Task, bud *budget.Budget) error {
 	var err error
 
-	//     find claims with lifecycle "propose" that are expired
-	var cla []*poststorage.Object
+	// find trees with lifecycle "propose" that are expired
+	var trees []*poststorage.Object
 	{
-		cla, err = h.sto.Post().SearchExpiry()
+		trees, err = h.sto.Post().SearchExpiry()
 		if err != nil {
 			return tracer.Mask(err)
 		}
 	}
 
-	for _, x := range cla {
-		var tre *big.Int
+	for _, x := range trees {
+		var treeId *big.Int
 		{
-			tre, err = metToTre(x.Meta)
+			treeId, err = metaToTreeId(x.Meta)
 			if err != nil {
 				return tracer.Mask(err)
 			}
 		}
 
-		var det [4]marketscontract.IMarketsClaim
+		var claims [4]marketscontract.IMarketsClaim
 		{
-			det, err = h.mar.Claims(nil, tre)
+			claims, err = h.markets.Claims(nil, treeId)
 			if err != nil {
 				return tracer.Mask(err)
 			}
 		}
 
-		fmt.Printf("det %#v\n", det)
+		var claimsLength *big.Int
```

Comment:
As discussed this whole call to fetch the claims length is probably not necessary here. We have the claims list already and can simply iterate to find the latest claim that is not `nil`.

Reply:
https://github.com/uvio-network/apiserver/pull/49/commits/a36e339ff97ac61bc23ad2a22f5619584e625ab1